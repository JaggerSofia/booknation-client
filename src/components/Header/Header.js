import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import TokenService from '../../services/token-service'
import './Header.css'

export default class Header extends Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
  }

  

  render() {
    return <>
      <nav className='Header'>
        <h1>
          <Link to='/'>
            <FontAwesomeIcon className='blue' />
            {' '}
            BookNation
          </Link>
        </h1>
        <span className='Header__tagline--wide'>Reviews by book lovers</span>

      </nav>

      <span className='Header__tagline--narrow'>Rate all the books.</span>
    </>
  }
}