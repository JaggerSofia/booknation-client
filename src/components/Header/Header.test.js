import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe(`Header component`, () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Header />, div);
      ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the component as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <Header />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
