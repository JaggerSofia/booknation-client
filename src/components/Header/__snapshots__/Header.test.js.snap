// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the component as expected 1`] = `
Array [
  <nav
    className="Header"
  >
    <h1>
      <a
        href="/"
        onClick={[Function]}
      >
         
        BookNation
      </a>
    </h1>
    <span
      className="Header__tagline--wide"
    >
      Reviews by book lovers
    </span>
    <div
      className="Header__not-logged-in"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Log in
      </a>
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </nav>,
  <span
    className="Header__tagline--narrow"
  >
    Rate all the books.
  </span>,
]
`;
